client/client.go

```diff
@@ -1225,31 +1223,39 @@ func (c *Client) updateNodeFromFingerprint(response *fingerprint.FingerprintResp
 	return c.configCopy.Node
 }
 
-// resourcesAreEqual is a temporary function to compare whether resources are
-// equal. We can use this until we change fingerprinters to set pointers on a
-// return type.
-func resourcesAreEqual(first, second *structs.Resources) bool {
-	if first.CPU != second.CPU {
+// updateNodeResources modifies the client NodeResources, returns whether it did so
+func updateNodeResources(c *config.Config, up *structs.NodeResources) bool {
+	if up == nil {
 		return false
 	}
-	if first.MemoryMB != second.MemoryMB {
-		return false
+	if c.NetworkInterface != "" || c.NetworkSpeed != 0 {
```

Comment:
Is it possible to manually specify a NetworkSpeed but apply it to a fingerprinted interface? I think it is and this code would break that?

Either way we should add a test asserting the existing behavior and make sure we maintain it (can probably be a unit test against updateNodeResources).

Reply:
you're right! I redid it and added tests